### PR TITLE
refactor(pie-switch): DSW-2280 use classes instead of attributes for styling

### PR DESCRIPTION
**Pull request by xander-marjoram**
## Author Checklist (complete before requesting a review)
- [x] I have performed a self-review of my code
- [x] I have reviewed the `PIE Storybook` PR preview
- [x] If changes will affect consumers of the package, I have created a changeset entry.
- [x] If a changeset file has been created, I have used the `/snapit` functionality to test my changes in a consuming application

### Reviewer 1 - @jamieomaguire 
- [x] I have reviewed the `PIE Storybook` PR preview

### Reviewer 2
- [x] I have reviewed the `PIE Storybook` PR preview

### CODE REVIEW

File: packages/components/pie-switch/src/index.ts [hunk 1]
```diff
@@ -22,34 +24,15 @@ const componentSelector = 'pie-switch';
  * @event {CustomEvent} change - when the switch checked state is changed.
  */
 export class PieSwitch extends FormControlMixin(RtlMixin(LitElement)) implements SwitchProps, PIEInputElement {
-    @query('input[type="checkbox"]')
-    private input?: HTMLInputElement;
-
-    protected firstUpdated (_changedProperties: PropertyValues<this>): void {
```
[the remark this answers is not in the document]
Comment (xander-marjoram) [reply packages/components/pie-switch/src/index.ts:28]: I also removed unnecessary calls to `super`

File: packages/components/pie-switch/src/switch.scss [hunk 2]
```diff
@@ -111,6 +111,12 @@
     background-color: var(--dt-color-interactive-light);
     padding: var(--switch-padding);
 
+    .c-switch-wrapper--rtl & {
```
Comment (xander-marjoram) [inline packages/components/pie-switch/src/switch.scss:114]: This had to be moved up to fix a `no-descending-specificity` linting error

File: packages/components/pie-switch/test/component/pie-switch.spec.ts [hunk 3]
```diff
@@ -19,12 +18,7 @@ test.describe('Component: `Pie switch`', () => {
 
     test('should be visible', async ({ mount, page }) => {
         // Arrange
-        await mount(PieSwitch, {
-            props: {
```
Comment (xander-marjoram) [inline packages/components/pie-switch/test/component/pie-switch.spec.ts:23]: These are default values anyway

**Review by jamieomaguire — APPROVED** [review]

**Review by JoshuaNg2332 — APPROVED** [review]

**Pull request merged**